upb/json_encode.c

```diff
@@ -162,20 +166,21 @@ static void jsonenc_timestamp(jsonenc* e, const upb_Message* msg,
    * Fliegel, H. F., and Van Flandern, T. C., "A Machine Algorithm for
    *   Processing Calendar Dates," Communications of the Association of
    *   Computing Machines, vol. 11 (1968), p. 657.  */
-  L = (int)(seconds / 86400) + 68569 + 2440588;
+  L = seconds / 86400.0 + 68569 + 2440588;
```

Comment:
This doesn't seem correct. I think the right fix here is to make the division to always round down, instead of rounding towards zero. We can do this by ensuring that seconds is always positive.

What if we change this to something like:

```
  seconds += 62135596800;  // Ensure seconds is positive.
  L = seconds / 86400 - 719162 + 68569 + 2440588
```

If seconds is always positive, then the `safe_mod()` changes below will no longer be necessary either.

Reply:
Perfect solution. I hadn't thought to keep the seconds positive.

Reply:
Perfect solution. I hadn't thought to keep the always positive.